docs: Add documentation for dotnet project convert command

devlead opened the pull request:
Looked at https://learn.microsoft.com/en-us/dotnet/core/tools/ and it would seem `dotnet project convert` was missing.

This PR proposes to add documentation for the dotnet project convert command, which converts file-based programs to project-based programs. 

The documentation includes:
- Command synopsis and description
- Conversion process details
- All command options (--dry-run, --force, --interactive, --output)
- Examples showing conversion from file-based to project-based structure
- Translation of #:sdk, #:package, and #:property directives


<!-- PREVIEW-TABLE-START -->

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/tools/dotnet-project-convert.md](https://github.com/dotnet/docs/blob/c69ef43a7cdbab9bf03311eb9815d7f72995ea94/docs/core/tools/dotnet-project-convert.md) | [docs/core/tools/dotnet-project-convert](https://review.learn.microsoft.com/en-us/dotnet/core/tools/dotnet-project-convert?branch=pr-en-us-50125) |
| [docs/core/tools/index.md](https://github.com/dotnet/docs/blob/c69ef43a7cdbab9bf03311eb9815d7f72995ea94/docs/core/tools/index.md) | [docs/core/tools/index](https://review.learn.microsoft.com/en-us/dotnet/core/tools/index?branch=pr-en-us-50125) |
| [docs/navigate/tools-diagnostics/toc.yml](https://github.com/dotnet/docs/blob/c69ef43a7cdbab9bf03311eb9815d7f72995ea94/docs/navigate/tools-diagnostics/toc.yml) | [docs/navigate/tools-diagnostics/toc](https://review.learn.microsoft.com/en-us/dotnet/navigate/tools-diagnostics/toc?branch=pr-en-us-50125) |


<!-- PREVIEW-TABLE-END -->

devlead committed:
docs: Add documentation for dotnet project convert command

Adds documentation for the dotnet project convert command, which converts
file-based programs to project-based programs. The documentation includes:

- Command synopsis and description
- Conversion process details
- All command options (--dry-run, --force, --interactive, --output)
- Examples showing conversion from file-based to project-based structure
- Translation of #:sdk, #:package, and #:property directives

devlead committed:
Address PR feedback

devlead committed:
Address PR feedback

meaghanlewis committed:
Remove empty line in dotnet-project-convert.md

meaghanlewis reviewed (commented):
@devlead thanks for adding this doc!

Can you also add an entry for it in: [the TOC](https://github.com/dotnet/docs/blob/main/docs/navigate/tools-diagnostics/toc.yml), and [this landing page](https://github.com/dotnet/docs/blob/main/docs/core/tools/index.md)?

devlead committed:
docs: Add dotnet project convert to navigation and index

Adds the dotnet project convert command to the CLI index and table of contents.

devlead committed:
Address PR feedback
